LayoutTests/editing/selection/ios/scrolling-after-caret-selection-inside-contenteditable-div.html

```diff
@@ -0,0 +1,116 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true focusStartsInputSessionPolicy=disallow AsyncOverflowScrollingEnabled=true ] -->
+<html>
+<head>
+	<meta charset="utf-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+	<script src="../../../resources/js-test.js"></script>
+	<script src="../../../resources/basic-gestures.js"></script>
+	<script src="../../../resources/ui-helper.js"></script>
+	<style>
+		html, body {
+			padding: 0;
+			margin: 0;
+		}
+
+		#editable {
+			display: inline-block;
+			border: 1px solid black;
+			margin: 2px;
+		}
+
+		#button {
+			font-size: 16px;
+			margin-top: 20px;
+			display: inline;
+		}
+
+		#overflow_container {
+			height: 300px;
+			overflow: scroll;
+			border: 2px solid black;
+		}
+
+		#overflow {
+			background: linear-gradient(180deg, rgba(255, 199, 153, 1) 0%, rgba(255, 0, 102, 1) 100%);
+			height: 600px;
+		}
+	</style>
+</head>
+<script>
+	jsTestIsAsync = true;
+
+	addEventListener("load", async () => {
+		var editableDivClicked = false;
+		var buttonClicked = false;
+
+		var fromX = overflow_container.offsetLeft + 50;
+		var fromY = overflow_container.offsetTop + overflow_container.offsetHeight - 5;
+		var toX = fromX;
+		var toY = fromY - 100;
+
+		var initialScrollPosition = overflow_container.scrollTop;
+
+		editable.addEventListener("click", (e) => {
+			editableDivClicked = true;
+			button.addEventListener("click", (e) => {
+				buttonClicked = true;
+			});
+		});
+
+		async function finish(passed, msg) {
+			overflow_container.remove();
+			editable.remove();
+			button.remove();
+			if (window.testRunner) {
+				if (passed)
+					testPassed(msg);
+				else
+					testFailed(msg);
+				
+				finishJSTest();
+			} else {
+				let result = passed ? "PASS: " : "FAIL: ";
+				output.innerText = result + msg;
+			}
+		}
+
+		if (window.testRunner) {
+			description(`When a site programmatically focuses a contenteditable div before editing or text selections 
+			             are made, and that div is then edited followed by editing ending in a manner which preserves 
+						 the text selection (i.e clicking a button element), scrollable areas should remain scrollable.`);
```

Comment:
Nit - it looks like there are a mix of tabs and spaces here.

I believe we also prefer 4-space indents, rather than the ObjC-style indentation that aligns the next line to the function call.
```suggestion
            description(`When a site programmatically focuses a contenteditable div before editing or text selections 
                are made, and that div is then edited followed by editing ending in a manner which preserves 
                the text selection (i.e clicking a button element), scrollable areas should remain scrollable.`);
```